Core/GDCore/Extensions/Metadata/AbstractFunctionMetadata.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <functional>
+#include <map>
+#include <memory>
+
+#include "GDCore/Events/Instruction.h"
+#include "GDCore/String.h"
+#include "ParameterMetadata.h"
+#include "ParameterOptions.h"
+
+namespace gd {
+class Project;
+class Layout;
+class EventsCodeGenerator;
+class EventsCodeGenerationContext;
+class SerializerElement;
+} // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Describe user-friendly information about an instruction (action or
+ * condition), its parameters and the function name as well as other information
+ * for code generation.
+ *
+ * \ingroup Events
+ */
+class GD_CORE_API AbstractFunctionMetadata {
```

Comment:
C++ needed more precised type declarations. This class is replacing the `InstructionMetadata | ExpressionMetadata | MultipleInstructionMetadata`.